roles/deploy/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+- name: Initialize
+  deploy_helper: path="{{ project_root }}" state=present
+
+- name: Clone project files
+  git: repo="{{ project_git_repo }}"
+       dest="{{ project_source_path }}"
+       version="{{ project_version }}"
+       accept_hostkey=yes
+
+- name: write unfinished file
+  file: path="{{ project_source_path }}/{{ deploy_helper.unfinished_filename }}" state=touch
+
+- name: Copy files to new build dir
+  command: cp -pr {{ project_source_path }} {{ deploy_helper.new_release_path }}
+
+- name: Move site into root
+  shell: mv {{ deploy_helper.new_release_path }}/{{ project_subtree }}/* {{ deploy_helper.new_release_path }}
+  when: project_subtree is defined
+
+- name: Remove unwanted files/folders from new release
+  file: path="{{ deploy_helper.new_release_path }}/{{ item }}" state=absent
+  with_items: project_unwanted_items
+
+- name: Copy project files
+  copy: src="{{ item.src }}" dest="{{ deploy_helper.new_release_path }}/{{ item.dest }}" mode="{{ item.mode | default('0644') }}"
+  with_items: project_files
```

Comment:
#### Optional compile locally
This works for me, right before the "Copy project files" task.
```yaml
- name: Run pre_build_commands_local on Ansible host
  local_action: "command {{ item.cmd }}"
  args:
    chdir: "{{ item.path }}"
  with_items: project_pre_build_commands_local
```
...with the following example definition in defaults or `wordpress_sites`
```yaml
project_pre_build_commands_local:
  - path: ../example.dev/web/app/themes/sage
    cmd: gulp --production
```
#### Switch "Copy project files" to rsync
I'd like to switch "Copy project files" to [synchronize](http://docs.ansible.com/synchronize_module.html) to use its `.rsync-filter`:
```yaml
- name: Copy project files
  synchronize: src="{{ item.src }}" dest="{{ deploy_helper.new_release_path }}/{{ item.dest }}"
  with_items: project.sync_files  | default([])
```
...used with this optional specification in `wordpress_sites`:
```yaml
wordpress_sites:
  example.com:
    sync_files:
      - name: compiled_theme_assets
        src: ../example.dev/web/app/themes/sage/dist
        dest: web/app/themes/sage
``` 
@mAAdhaTTah this could replace the rsync task discussed earlier.